### ISSUE TITLE
[Codabox Connect] Invalid payload when searching documents

### ISSUE THREAD

**Issue by thomasvdb** [ibanity/ibanity-dotnet#3]
We're currently trying to integrate with Codabox Connect but when doing a document search we get the following response:

```
Unhandled exception. Ibanity.Apis.Client.Http.IbanityClientException: Request 0001B2779489D1030A01069401BB63DA3DEF1821DC1AFD failed (400 BadRequest): invalidPayload (The submitted payload could not be parsed. Unknown property: meta)
   at Ibanity.Apis.Client.Http.HttpResponseMessageExtensions.ThrowOnFailure(HttpResponseMessage this, ISerializer`1 serializer, ILogger logger)
   at Ibanity.Apis.Client.Http.ApiClient.SendWithPayload[TRequest,TResponse](HttpMethod method, String path, String bearerToken, TRequest payload, Nullable`1 idempotencyKey, CancellationToken cancellationToken)
   at Ibanity.Apis.Client.Products.BaseResourceClient`6.InternalCreate[TRequestAttributes,TRequestMeta,TRequestRelationships,TRequestLinks](TToken token, String path, Data`4 payload, Nullable`1 idempotencyKey, Nullable`1 cancellationToken)
   at Program.<Main>$(String[] args) in /Users/thomasvdb/Research/IbanityClientResearch/IbanityClientResearch/Program.cs:line 29
   at Program.<Main>(String[] args)
```

This is the code of a Console program I tried setting up:

```
using Ibanity.Apis.Client;
using Ibanity.Apis.Client.Products.CodaboxConnect.Models;

var certificatePath = "";
var certificatePassword = "";
var codaboxConnectClientId = "";
var codaboxConnectClientSecret = "";
var consentId = Guid.Parse("");
var customerCompanyNumber = "";

var ibanityService = new IbanityServiceBuilder().
    SetEndpoint("https://api.ibanity.com").
    AddClientCertificate(
        certificatePath,
        certificatePassword).
    AddCodaboxConnectOAuth2Authentication(codaboxConnectClientId, codaboxConnectClientSecret).
    Build();

var token = await ibanityService.CodaboxConnect.ClientTokenService.GetToken();
var consent = await ibanityService.CodaboxConnect.AccountingOfficeConsents.Get(token, consentId);

Console.WriteLine($"Consent found. Accounting office ID: {consent.AccountingOfficeId}");

var documentSearchRequest = new DocumentSearch
{
    DocumentType = "creditCardStatement", 
    From = new DateTimeOffset(2022, 1, 1, 0, 0, 0, TimeSpan.Zero),
    To = new DateTimeOffset(2023, 1, 1, 0, 0, 0, TimeSpan.Zero),
};
        
var documentSearchResponse = await ibanityService.CodaboxConnect.DocumentSearches.Create(
    token, 
    consent.AccountingOfficeId, 
    documentSearchRequest,
    new [] { customerCompanyNumber }
);

foreach (var document in documentSearchResponse.Documents)
{
    Console.WriteLine($"Document ID: {document.Id}");
}
```

Fetching the consent works but the document search fails with the exception mentioned in the beginning.
Is there something incorrectly configured in the request?

Tested this with .NET 6 on macOS Ventura 13.2

**Comment by stephdewit** [comment]
Hello @thomasvdb

Could you share with us the message of the exception you received? It should contain a request ID which will help us diagnosing the issue.

Thanks!

**Comment by stephdewit** [comment]
Hello @thomasvdb

It was caused by an issue in the library. We released a 0.5.3 version fixing this.

Your issue should be fixed after upgrading to the new NuGet package version.

Feel free to reopen this issue if you still encounter the problem.
